.evergreen/compile-scan-build.sh

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o pipefail
+
+: "${BSONCXX_POLYFILL:?}"
+: "${CXX_STANDARD:?}"
+
+mongoc_prefix="$(pwd)/../mongoc"
+
+# shellcheck source=/dev/null
+. "${mongoc_prefix:?}/.evergreen/scripts/find-cmake-latest.sh"
+export cmake_binary
+cmake_binary="$(find_cmake_latest)"
+command -v "$cmake_binary"
+
+# scan-build binary is available in different locations depending on the distro.
+# Search for a match in order of preference as listed.
+declare -a scan_build_directories
+
+scan_build_directories+=(
+  # Prefer toolchain scan-build if available.
+  "/opt/mongodbtoolchain/v4/bin"
+  "/opt/mongodbtoolchain/v3/bin"
+
+  # Use system scan-build otherwise.
+  "/usr/bin"
+)
```

Comment:
Recommend consulting `$PATH` for other locations.
```suggestion
IFS=: read -ra sys_path <<< "$PATH"
scan_build_directories+=(
  # Prefer toolchain scan-build if available.
  "/opt/mongodbtoolchain/v4/bin"
  "/opt/mongodbtoolchain/v3/bin"

  # Use system scan-build otherwise.
  "${sys_path[@]}"
)
```